autogen/agentchat/groupchat.py

```diff
@@ -660,6 +666,12 @@ def clear_agents_history(self, reply: str, groupchat: GroupChat) -> str:
                     elif agent.name == word[:-1]:  # for the case when agent name is followed by dot or other sign
                         agent_to_memory_clear = agent
                         break
+        # preserve last tool call message if function inside of tool response
+        if "tool_responses" in reply and not nr_messages_to_preserve:
```

Comment:
When would this happen? If "tool_responses" is in reply, would "CLEAR HISTORY" be triggered? Is there a test covering this case?

Reply:
oh, thaere was not descriptive commentary; I changed it. That case happens when we calling "clear history" without `nr_messages_to_preserve `argument from inside of function response. Before that caused remove of tool call while tool response left. 

Created also test case for scenario.